pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java

```diff
@@ -211,7 +211,7 @@ public CompletableFuture<Boolean> canConsumeAsync(TopicName topicName, String ro
             if (isSuperUser) {
                 return CompletableFuture.completedFuture(true);
             } else {
-                return provider.canConsumeAsync(topicName, role, authenticationData, subscription);
+                return provider.allowTopicOperationAsync(topicName, role, TopicOperation.CONSUME, authenticationData);
```

Comment:
1. Please remove the superuser check, because the `provider.allowTopicOperationAsync` includes that.
2. Please pass the `subscription` by the `new AuthenticationDataSubscription(authenticationData, subscription)`

Reply:
Fixed.